### PR TITLE
Constructors for the lights. Issue #297

### PR DESCRIPTION
**Pull request by zzuegg**

### CODE REVIEW

**Comment by Nehon** [comment]
Thanks for this.
But you need to add the empty constructors too.

**Comment by zzuegg** [comment]
Added the missing default empty constructors

**Comment by DannyJo** [comment]
Nice stuff. Might not be a big deal but in the second constructor in SpotLight.java you get double calls to computeAngleParameters().

**Comment by Nehon** [comment]
Yup, if you remove this this PR is ready to be merged.

File: jme3-core/src/main/java/com/jme3/light/SpotLight.java [hunk 1]
```diff
@@ -74,6 +75,39 @@ public SpotLight() {
...
+     * @param position Position of the light
+     * @param direction Direction of the light
+     * @param spotInnerAngle Inner angle of the light
+     * @param spotOuterAngle Outer angle of the light
+     * @param spotRange Range of the light
+     */
+    public SpotLight(ColorRGBA lightColor,Vector3f position, Vector3f direction, float spotInnerAngle, float spotOuterAngle,float spotRange) {
+        this(lightColor,position,direction,spotInnerAngle,spotOuterAngle);
+        setSpotRange(spotRange);
+        computeAngleParameters();
```
Comment (Nehon) [inline jme3-core/src/main/java/com/jme3/light/SpotLight.java]: This call to computeAngleParameters(); can be removed as it's been called in the prior thi(...) call

File: jme3-core/src/main/java/com/jme3/light/SpotLight.java [hunk 2]
```diff
@@ -58,19 +59,50 @@
...
+    protected float spotOuterAngle;
+    protected float spotRange;
+    protected float invSpotRange;
+    protected float packedAngleCos;
     
     protected float outerAngleCosSqr, outerAngleSinSqr;
     protected float outerAngleSinRcp, outerAngleSin, outerAngleCos;
     
     public SpotLight() {
+        this(new ColorRGBA(1f, 1f, 1f, 1f), new Vector3f(), new Vector3f(0, -1, 0), FastMath.QUARTER_PI / 8, FastMath.QUARTER_PI / 6);
```
Comment (Shredder121) [inline jme3-core/src/main/java/com/jme3/light/SpotLight.java]: Can't these calculations be put in a static final field or something?

Comment (empirephoenix) [inline jme3-core/src/main/java/com/jme3/light/SpotLight.java]: Why? that would bloat the class, and the calculations need no real time at all. It is not that you are creating 1 million lights per frame. And if you want, you can simply write yourself a factory class using the second constructor, that has precalculated values in the factory.

Comment (Shredder121) [inline jme3-core/src/main/java/com/jme3/light/SpotLight.java]: Okay then, just wanted to add my 2 cents.

Comment (zzuegg) [inline jme3-core/src/main/java/com/jme3/light/SpotLight.java]: Actually, if you submit your own color/direction, you are saving 2 object creations compared with the previous version. 
+The divisions should be resolved during compile time.

**Comment by Nehon** [comment]
Ok back to this.
I'm a bit annoyed because there are 2 PRs that does the same thing see #306.
We've been bad at talking to each others I guess. That's just too bad that the work has been done twice.
So since I'm going to integrate one or the other I have to choose and I'm gonna choose the one that will give me the less work because I'm a lazy ass.
I'm gonna choose PR #306 because it's a tiny bit cleaner to my taste, and it even fixes my lousy spelling in the java doc.
However it doesn't have the constructor with the radius for PointLight and the spot angles for the SpotLight...so mixed feelings, I'll add the 2 constructors.
So @zzuegg I'm really sorry to have wasted your time. Tell me how I can make it up to you ;)

**Pull request closed without merging**